apps/site/components/Common/Search/index.tsx

```diff
@@ -1,61 +1,64 @@
-import { useTranslations } from 'next-intl';
-import { useState, type FC } from 'react';
+import { OramaSearchBox, OramaSearchButton } from '@orama/react-components';
+import { useTheme } from 'next-themes';
+import { type FC } from 'react';
 
-import { WithSearchBox } from '@/components/Common/Search/States/WithSearchBox';
-import { useDetectOS } from '@/hooks';
-import { useKeyboardCommands } from '@/hooks/react-client';
+import {
+  ORAMA_CLOUD_ENDPOINT,
+  ORAMA_CLOUD_API_KEY,
+  DEFAULT_ORAMA_QUERY_PARAMS,
+  BASE_URL,
+} from '@/next.constants.mjs';
 
-import styles from './index.module.css';
+import { themeConfig } from './utils';
 
-export const SearchButton: FC = () => {
-  const [isOpen, setIsOpen] = useState(false);
-  const t = useTranslations();
-  const openSearchBox = () => setIsOpen(true);
-  const closeSearchBox = () => setIsOpen(false);
-
-  useKeyboardCommands(cmd => {
-    switch (cmd) {
-      case 'cmd-k':
-        openSearchBox();
-        break;
-      case 'escape':
-        closeSearchBox();
-        break;
-      default:
-    }
-  });
-
-  const { os } = useDetectOS();
-
-  const osCommandKey = os === 'MAC' ? '⌘' : 'Ctrl';
-  const isOSLoading = os === 'LOADING';
+const SearchButton: FC = () => {
+  const { resolvedTheme } = useTheme();
+  const colorScheme = resolvedTheme as 'light' | 'dark';
 
   return (
     <>
-      <button
-        type="button"
-        onClick={openSearchBox}
-        className={styles.searchButton}
-        aria-label={t('components.search.searchBox.placeholder')}
+      <OramaSearchButton
+        style={{ flexGrow: 1 }}
+        colorScheme={colorScheme}
+        themeConfig={themeConfig}
       >
-        <MagnifyingGlassIcon className={styles.magnifyingGlassIcon} />
+        Search
+      </OramaSearchButton>
 
-        {t('components.search.searchBox.placeholder')}
-        <kbd
-          title={`${osCommandKey} K`}
-          className={classNames(styles.shortcutIndicator, {
-            'opacity-0': isOSLoading,
-          })}
-        >
-          <abbr>{osCommandKey} K</abbr>
-        </kbd>
-      </button>
-
-      {isOpen ? <WithSearchBox onClose={closeSearchBox} /> : null}
+      <OramaSearchBox
+        index={{ api_key: ORAMA_CLOUD_API_KEY, endpoint: ORAMA_CLOUD_ENDPOINT }}
+        colorScheme={colorScheme}
+        themeConfig={themeConfig}
+        facetProperty="siteSection"
+        sourceBaseUrl={BASE_URL}
+        sourcesMap={{
+          title: 'pageSectionTitle',
+          description: 'formattedPath',
+          path: 'pageLink',
+        }}
+        resultMap={{
+          title: 'pageSectionTitle',
+          description: 'formattedPath',
+          section: 'siteSection',
+          path: 'pageLink',
+        }}
+        highlight={{
+          caseSensitive: false,
+          HTMLTag: 'b',
+          CSSClass: 'font-bold',
+        }}
+        searchParams={DEFAULT_ORAMA_QUERY_PARAMS}
+        linksTarget="_self"
+        suggestions={[
```

Comment:
Can this not be hardcoded? 🤔 Or at least be a JSON file which we can configure?

Reply:
Your call! As long as this is an array of strings, this can come from anywhere. Right now we don't expose a way to edit this from the Orama Cloud dashboard (will do in the future).

Reply:
Sure thing, I will add a configurable constant like DEFAULT_ORAMA_QUERY_PARAMS

Reply:
Pro tip: Can Orama automatically infer this based on most searched results? :)

Reply:
this will be available soon. for now, we still need to pass the first suggestions in the empty state as "hard-coded". However, when available, Orama already generates the follow-up question suggestions, automatically (based on the context of the previous interaction), but this doesn't apply for the first suggestions yet.